invokeai/app/services/item_storage/item_storage_memory.py

```diff
@@ -0,0 +1,28 @@
+from typing import Generic, Optional, TypeVar
+
+from pydantic import BaseModel
+
+from invokeai.app.services.item_storage.item_storage_base import ItemStorageABC
+
+T = TypeVar("T", bound=BaseModel)
+
+
+class ItemStorageMemory(ItemStorageABC, Generic[T]):
+    def __init__(self, id_field: str = "id") -> None:
+        super().__init__()
+        self._id_field = id_field
+        self._items: dict[str, T] = {}
+
+    def get(self, item_id: str) -> Optional[T]:
+        return self._items.get(item_id)
+
+    def set(self, item: T) -> None:
```

Comment:
Is there any sort of cleanup at play here? Won't the dict in memory get really big until the user restarts their server?

Reply:
Whoops! I had a note for myself to do this but forgot. I've implemented `max_size` in two ways:

- With an unordered dict, no guarantees about the order of items: [e49cd87](https://github.com/invoke-ai/InvokeAI/pull/5603/commits/e49cd874879a5eb182e3cb1e9a6e57ec23d2f7c0)
- With an `OrderedDict`, using an LRU eviction strategy: [1d7bc01](https://github.com/invoke-ai/InvokeAI/pull/5603/commits/1d7bc01d138b3b47f02fba09c3b2d89a438cc07a)

I think the latter is the better option.

I also added tests for this both implementations.